packages/cms-editor/src/main/control/AddContentObject.tsx

```diff
@@ -0,0 +1,149 @@
+import type { CmsData, ContentObject, MapStringString } from '@axonivy/cms-editor-protocol';
+import {
+  BasicField,
+  Button,
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+  Flex,
+  hotkeyText,
+  Input,
+  selectRow,
+  Textarea,
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+  useHotkeys
+} from '@axonivy/ui-components';
+import { IvyIcons } from '@axonivy/ui-icons';
+import { useQueryClient } from '@tanstack/react-query';
+import { type Table } from '@tanstack/react-table';
+import { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useAppContext } from '../../context/AppContext';
+import { useClient } from '../../protocol/ClientContextProvider';
+import { useMeta } from '../../protocol/use-meta';
+import { genQueryKey, useQueryKeys } from '../../query/query-client';
+import { useKnownHotkeys } from '../../utils/hotkeys';
+import { useClientLanguage } from '../../utils/use-client-language';
+
+type AddContentObjectProps = {
+  table: Table<ContentObject>;
+};
+
+export const AddContentObject = ({ table }: AddContentObjectProps) => {
+  const { t } = useTranslation();
+  const { context, contentObjects, selectedContentObject, setSelectedContentObject } = useAppContext();
+
+  const [open, setOpen] = useState(false);
+  const onOpenChange = (open: boolean) => {
+    setOpen(open);
+    if (open) {
+      initializeDialog();
+    }
+  };
+
+  const [name, setName] = useState('');
+  const [namespace, setNamespace] = useState('');
+  const [values, setValues] = useState<MapStringString>({});
+
+  const locales = useMeta('meta/locales', context, []).data;
+  const { clientLanguageTag, languageDisplayName } = useClientLanguage();
+  const isClientLanguageInCms = useMemo(() => locales.includes(clientLanguageTag), [clientLanguageTag, locales]);
+
+  const initializeDialog = () => {
+    setName('NewContentObject');
+    setNamespace(initialNamespace(contentObjects, selectedContentObject));
+    setValues(isClientLanguageInCms ? { [clientLanguageTag]: '' } : {});
+  };
+
+  const changeValue = (languageTag: string, value: string) => {
+    const newValues = structuredClone(values);
+    newValues[languageTag] = value;
+    setValues(newValues);
+  };
+
+  const client = useClient();
+  const queryClient = useQueryClient();
+  const { dataKey } = useQueryKeys();
+  const addContentObject = (event: React.MouseEvent<HTMLButtonElement> | KeyboardEvent) => {
+    const uri = `${namespace}/${name}`;
+    client.create({ context, contentObject: { uri, type: 'STRING', values } });
+    queryClient.invalidateQueries({ queryKey: genQueryKey('data') }).then(() => {
```

Comment:
maybe think about wrap this into a react query mutation function. then you would also get pending states etc.